File: ModernCocoaFarmerBindingsLibrary/Enums/ComparisonOperator.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Text;

namespace MCF.Logic
{
    public enum ComparisonOperator
    {
        [Description("Less Than")]
        LessThan,

        [Description("Less Than Or Equal To")]
        LessThanOrEqualTo,

        [Description("Equals")]
        Equals,

        [Description("Greater Than Or Equal To")]
        GreaterThanOrEqualTo,

        [Description("Greater Than")]
        GreaterThan
    }
}

File: ModernCocoaFarmer/Source/Logic/ComparisonOperator.cpp
#include "Logic/ComparisonOperator.h"
#include "Assert/Assert.h"


namespace Celeste
{
  //------------------------------------------------------------------------------------------------
  template <>
  std::string to_string(MCF::Logic::ComparisonOperator comparisonOperator)
  {
    using namespace MCF::Logic;

    switch (comparisonOperator)
    {
    case ComparisonOperator::kLessThan:
      return "LessThan";

    case ComparisonOperator::kLessThanOrEqualTo:
      return "LessThanOrEqualTo";

    case ComparisonOperator::kEqual:
      return "Equals";

    case ComparisonOperator::kGreaterThanOrEqualTo:
      return "GreaterThanOrEqualTo";

    case ComparisonOperator::kGreaterThan:
      return "GreaterThan";

    default:
      ASSERT_FAIL();
      return "";
    }
  }
}